doc/man7/EVP_PKEY-ML-DSA.pod

```diff
@@ -0,0 +1,114 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY-ML-DSA, EVP_KEYMGMT-ML-DSA,
+EVP_PKEY-ML-DSA-44, EVP_PKEY-ML-DSA-65, EVP_PKEY-ML-DSA-87
+- EVP_PKEY ML-DSA keytype and algorithm support
+
+=head1 DESCRIPTION
+
+ML-DSA implements the algorithms B<ML-DSA-44>, B<ML-DSA-65> and B<ML-DSA-87>. 
+The key types B<EVP_PKEY_ML_DSA_44>, B<EVP_PKEY_ML_DSA_65> and
+B<EVP_PKEY_ML_DSA_87> are implemented in OpenSSL's default provider.
+These implementations support the associated key, containing the public key I<pub>
+and the private key I<priv>.
+
+Each of the different key types has an associated security category.
+This value is one of 2, 3 or 5 for key types B<ML-DSA-44>, B<ML-DSA-65>
+and B<ML-DSA-87> respectively, which correspond to security strengths of
+128, 192 and 256 repsectively.
+
+=head2 Keygen Parameters
+
+=over 4
+
+=item "seed" (B<OSSL_PKEY_PARAM_ML_DSA_SEED>) <octet string>
+
+The seed used to generate the private and public key components in a
+deterministic manner. This should only be used for testing purposes.
+The length of the value supplied must be 32 bytes.
+When this value is not supplied the seed is generated randomly using a DRBG. 
+
+=item "properties" (B<OSSL_PKEY_PARAM_PROPERTIES>) <utf8_string>
+
+Sets properties to be used when fetching algorithm implementations used for
+ML-DSA hashing operations.
+
+=back
+
+Use EVP_PKEY_CTX_set_params() after calling EVP_PKEY_keygen_init().
+
+=head2 Common ML-DSA parameters
+
+In addition to the common parameters that all keytypes should support (see
+L<provider-keymgmt(7)/Common parameters>), the implementation of these key types
+support the following.
+
+The following parameters are gettable using EVP_PKEY_get_octet_string_param(),
+and settable when using EVP_PKEY_fromdata().
+
+=over 4
+
+=item "pub" (B<OSSL_PKEY_PARAM_PUB_KEY>) <octet string>
+
+The encoded public key value of size 1312, 1952 or 2592 bytes depending on the
+respective key type of B<ML-DSA-44>, B<ML-DSA-65> or B<ML-DSA-87>.
+
+=item "priv" (B<OSSL_PKEY_PARAM_PRIV_KEY>) <octet string>
+
+The encoded private key value of size 2560, 4032 or 4896 bytes depending on the
+respective key type of B<ML-DSA-44>, B<ML-DSA-65> or B<ML-DSA-87>.
```

Comment:
These are actually the *raw* public and private keys, but it just so happens that `"pub"` is also the encoded public key parameter, while there is no *encoded* private key parameter at this time. The raw forms are used in `export/import` and with some EVP "raw" key APIs, while the encoded form is used with `EVP_PKEY_set1_encoded_public_key()` via `EVP_PKEY_set_octet_string_param()` which is a `set` rather than `import`.

Reply:
I dont have an issue with what is written..
_PRIV_KEY and _PUB_KEY already imply raw key. I dotn think saying it is the encoded form is wrong here.

Reply:
You could twist my arm and hold firm, but I do think there's some value in helping the reader understand the distinct interfaces involved, so even when the "encoded" and "raw" forms are the same, we should (I think) strive to use the right language in context. The private key is always "raw", there is no parameter associated with an "encoded" form as distinct from "raw".